domain/keyupdater/state/state.go

```diff
@@ -46,97 +44,73 @@ FROM machine
 WHERE name = $machineName.name
 `, machineArg)
 	if err != nil {
-		return nil, fmt.Errorf(
-			"preparing select statement for getting machine %q when determining authorised keys: %w",
+		return errors.Errorf(
+			"preparing statement for checking if machine %q exists: %w",
 			name, err,
 		)
 	}
 
-	stmt, err := s.Prepare(`
-SELECT public_key AS &authorisedKey.*
-FROM user_public_ssh_key
-`, authorisedKey{})
-	if err != nil {
-		return nil, fmt.Errorf(
-			"preparing select statement for getting machine %q authorised keys: %w",
-			name, err,
-		)
-	}
-
-	authorisedKeys := []authorisedKey{}
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		// Because we have two queries to run and two error paths we need to
-		// handle the errors inside this TX so we can produce the correct error.
 		err := tx.Query(ctx, machineStmt, machineArg).Get(&machineArg)
 		if errors.Is(err, sqlair.ErrNoRows) {
-			return fmt.Errorf(
-				"cannot get authorised keys for machine %q: %w",
-				name, machineerrors.MachineNotFound,
-			)
+			return errors.Errorf(
+				"machine %q does not exist", name,
+			).Add(machineerrors.MachineNotFound)
 		} else if err != nil {
-			return fmt.Errorf(
-				"cannot get authorised keys for machine %q: %w",
-				name, err,
-			)
-		}
-
-		err = tx.Query(ctx, stmt).GetAll(&authorisedKeys)
-		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
-			return fmt.Errorf(
-				"cannot get authorised keys for machine %q: %w",
-				name, err,
+			return errors.Errorf(
+				"checking if machine %q exists: %w", name, err,
 			)
 		}
 		return nil
 	})
 
 	if err != nil {
-		return nil, err
-	}
-
-	rval := make([]string, 0, len(authorisedKeys))
-	for _, authKey := range authorisedKeys {
-		rval = append(rval, authKey.PublicKey)
+		return err
 	}
 
-	return rval, nil
+	return nil
 }
 
-// AllAuthorisedKeys returns all authorised keys for the model.
-func (s *State) AllAuthorisedKeys(ctx context.Context) ([]string, error) {
+// GetModelUUID returns the unique id for the model represented by this state.
```

Comment:
Now that we're using UUID, should the comment be "... returns the UUID for the model ..."

Reply:
Done & Done